File: Year-2022/LeetCode-July-Challenge/findFirstandLastPositionofElementinSortedArray.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] res = new int[]{-1,-1};
        if(nums.length == 0) return res;
        int start = 0, end = nums.length - 1;
        while(start < end){
            int mid = start + (end - start)/2;
            if(nums[mid] >= target) end = mid;
            else start = mid + 1;
        }
        if(nums[start] != target) return res;
        res[0] = start;
        end = nums.length;
        while(start < end){
            int mid = start + (end - start)/2;
            if(nums[mid] > target) end = mid;
            else start = mid + 1;
        }
        res[1] = start - 1;
        return res;
    }
}
public class findFirstandLastPositionofElementinSortedArray {
    public static void main(String[] args) {
        int[] nums = {5,7,7,8,8,10};
        int target = 8;
        Solution s = new Solution();
        int[] res = s.searchRange(nums, target);
        for(int i = 0; i < res.length; ++i) System.out.print(res[i] + " ");
    }
}

File: Year-2022/LeetCode-July-Challenge/findFirstandLastPositionofElementinSortedArray.js
var searchRange = function(nums, target) {
    let res = new Array(2);
    res.fill(-1);
    if(nums.length === 0) return res;
    let start = 0, end = nums.length - 1;
    while(start < end){
        let mid = Math.floor((start+end)/2);
        if(nums[mid] >= target) end = mid;
        else start = mid + 1;
    }
    if(nums[start] !== target) return res;
    res[0] = start;
    end = nums.length;
    while(start < end){
        let mid = Math.floor((start+end)/2);
        if(nums[mid] > target) end = mid;
        else start = mid + 1;
    }
    res[1] = start - 1;
    return res;
};
let nums = [5,7,7,8,8,10], target = 8;
console.log(searchRange(nums, target));